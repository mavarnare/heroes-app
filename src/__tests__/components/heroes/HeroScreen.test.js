import { mount } from "enzyme";
import { MemoryRouter, Route } from "react-router-dom";
import { HeroScreen } from "../../../components/heroes/HeroScreen";

describe('Tests for <HeroScreen />', () => {

    const history = {
        push: jest.fn(),
        goBack: jest.fn(),
        length: 10,
    };

    it('should render <Redirect /> if URL has no parameters', () => {
        const wrapper = mount(
            <MemoryRouter initialEntries={['/hero']}>
                <HeroScreen history={ history } />
            </MemoryRouter>
        );
        expect( wrapper.find('Redirect').exists() ).toBe( true );
    });
    

    it('should display a hero info if the parameter value is valid and exists', () => {
        const wrapper = mount(
            <MemoryRouter initialEntries={['/hero/marvel-spider']}>
                <Route exact path="/hero/:heroeId" component={ HeroScreen } />
            </MemoryRouter>
        );
        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('.row').exists() ).toBe( true );
    });

    it('should go back to previous screen using PUSH when history.length <= 2', () => {
        const history = {
            push: jest.fn(),
            goBack: jest.fn(),
            length: 1,
        };

        const wrapper = mount(
            <MemoryRouter initialEntries={['/hero/marvel-spider']}>
                <Route
                    path="/hero/:heroeId"
                    component={ () => <HeroScreen history={ history } /> }
                />
            </MemoryRouter>
        );
        wrapper.find('button').prop('onClick')();
        expect( history.push ).toHaveBeenCalledWith('/');
        expect( history.goBack ).not.toHaveBeenCalled();
    });

    it('should go back to previous screen using POP when history.length > 2', () => {
        const wrapper = mount(
            <MemoryRouter initialEntries={['/hero/marvel-spider']}>
                <Route
                    path="/hero/:heroeId"
                    component={ () => <HeroScreen history={ history } /> }
                />
            </MemoryRouter>
        );
        wrapper.find('button').prop('onClick')();
        expect( history.goBack ).toHaveBeenCalledTimes(1);
        expect( history.push ).not.toHaveBeenCalled();
    });

    it('should render <Redirect /> when URL parameters are not valid', () => {
        const wrapper = mount(
            <MemoryRouter initialEntries={['/hero/marvel-spider123']}>
                <Route
                    path="/hero/:heroeId"
                    component={ () => <HeroScreen history={ history } /> }
                />
            </MemoryRouter>
        );
        expect( wrapper.find('.row').exists() ).toBe( false );
        expect( wrapper.text() ).toBe('');
    });
})
