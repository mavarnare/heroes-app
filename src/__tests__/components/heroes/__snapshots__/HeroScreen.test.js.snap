// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for <HeroScreen /> should display a hero info if the parameter value is valid and exists 1`] = `
<div
  className="row mt-5"
>
  <div
    className="col-4"
  >
    <img
      alt="Spider Man"
      className="img-thumbnail animate__animated animate__fadeInLeft"
      src=""
    />
  </div>
  <div
    className="col-8 animate__animated animate__fadeIn"
  >
    <h3>
      Spider Man
    </h3>
    <ul
      className="list-group list-group-flush"
    >
      <li
        className="list-group-item"
      >
         
        <b>
          Alter ego:
        </b>
         
        Peter Parker
         
      </li>
      <li
        className="list-group-item"
      >
         
        <b>
          Publisher:
        </b>
         
        Marvel Comics
         
      </li>
      <li
        className="list-group-item"
      >
         
        <b>
          First appearance:
        </b>
         
        Amazing Fantasy #15
         
      </li>
    </ul>
    <h5>
       Characters 
    </h5>
    <p>
       
      Peter Parker
       
    </p>
    <button
      className="btn btn-outline-info"
      onClick={[Function]}
    >
      Return
    </button>
  </div>
</div>
`;
